web/src/routes/package.js

```diff
@@ -112,8 +120,111 @@ package_router.post('/', async (req,res) => {
                     }
                 }
             }
-            else { // ingestion
+            else { // Ingestion: user provided a URL
+                // check if the input is formatted correctly --> URL is either github or npm
+                url_elements = newPackageDataSchema.URL.split('/')
+                if( !(url_elements.includes('github.com') || url_elements.includes('www.npmjs.com')) )
+                {
+                    res.status(400).json({ message: 'There is missing fields in PackageData or the URL is formed improperly' })
+                }
+                // check if a package with that URL already exists
+                else if(await PackageData.findOne({ URL: newPackageDataSchema.URL })) {
+                    res.status(409).json({ message: 'Package exists already.' })
```

Comment:
This may need to be changed if we decide that the same package with different versions can be uploaded but for now I believe it is fine